pkg/virt-controller/watch/snapshot_base.go

```diff
@@ -381,9 +381,46 @@ func (ctrl *SnapshotController) handleVM(obj interface{}) {
 		for _, k := range keys {
 			ctrl.vmSnapshotQueue.Add(k)
 		}
+
+		ctrl.updateSnapshotStatuses(vm)
```

Comment:
Should add a new queue to handle these requests asynchronously.  Don't want to block this qoroutine.

Reply:
OK